javalib/src/main/scala/java/util/AbstractList.scala

```diff
@@ -253,11 +258,14 @@ private class RandomAccessListIterator[E](list: List[E], private var i: Int,
 
   def add(e: E): Unit = {
     list.add(i, e)
-    end += 1
+    onSizeChanged(1)
     last = -1
     i += 1
   }
 
+  protected def onSizeChanged(delta: Int): Unit =
+    end += delta
```

Comment:
Because in the views it is used as an event to notify that the size has changed.